### ISSUE TITLE
Investigate whether using a Docker volume or bind mount is more appropriate

### ISSUE THREAD

**Issue by jpdillingham** [slskd/slskd#439]
Doing a deep dive on out of space issues, I'm seeing that I have a Docker overlay volume that is pretty huge relative to the files that are on disk.  Furthermore, I'm getting out of space issues within the container while there's still some space on the physical disk.

The Dockerfile adds a volume, and I'm mapping this volume to a directory on the host.  It appears as though this mapping would be better achieved with a bind mount, to side-step issues with the overlay.  The Docker volume isn't buying much in this case, as almost nobody would use the container in such a way that nothing was mounted at all, and it doesn't seem like too much of a stretch to just force the entire app directory to be bound in this way.

It might be advantageous to create a mount for the `app` directory so that config, application data and logs can be persisted within a volume, and to move the incomplete and downloads directories back out of the app directory.  These directories would then not write to the volume, but instead either into the volatile area of the container, or (in 100% of use cases) out to a bind mount on the host.

It is also possible that users can achieve this themselves by redirecting the downloads and incomplete directories within config and bind-mounting them.  This should be the default, though.

**Comment by jpdillingham** [comment]
I'm mostly creating this issue as a note for myself to continue investigation later but I'd welcome comments from anyone that has noticed this behavior in their own environment.

Right now my (perhaps incorrect) understanding is that incoming files go:

`slskd -> /app directory -> /app volume -> host`

Where the existence of `/app volume` is causing data that should be a simple pass-through to be persisted in an overlay layer.

I think it is more correct to have incoming files go:

`slskd -> /app directory -> host` 

By way of a simple bind-mount.

**Comment by jpdillingham** [comment]
Unsurprisingly my understanding _was_ incorrect, and it appears that the `overlay` filesystems added by Docker just mirror the primary mount instead of containing data themselves.

My space problem was due to me forgetting that I had uploaded a bunch of files to share and then decided to move the shared directory without moving or deleting the files, so all this space that I thought Docker was cannibalizing was actually self inflicted.

Through reading up about volumes and how the engine filesystem works, it is clear that this sort of issue will arise if the end user fails to map the `app` volume when starting the container, and I think that's the intended experience so that data and files are non volatile.
